Makefile

```diff
@@ -1,8 +1,40 @@
+generate-xcodeproj:
+	swift package generate-xcodeproj
+
+xcodeproj: generate-xcodeproj
+	xed .
+
 test-linux:
 	docker build --tag parser-printer . \
 		&& docker run --rm parser-printer
 
 test-swift:
 	swift test
 
-test-all: test-linux test-macos test-ios
+test-macos: generate-xcodeproj
+	set -o pipefail && \
+	xcodebuild test \
+		-scheme ParserPrinter-Package \
+		-destination platform="macOS" \
+		-derivedDataPath ./.derivedData \
+		| xcpretty
+
+test-ios: generate-xcodeproj
+	set -o pipefail && \
+	xcodebuild test \
+		-scheme ParserPrinter-Package \
+		-destination platform="iOS Simulator,name=iPhone XR,OS=12.0" \
+		-derivedDataPath ./.derivedData \
+		| xcpretty
+
+test-playgrounds:
+	swift \
+		-F .derivedData/Build/Products/Debug/ \
```

Comment:
we can use the build artifacts from `xcodebuild` to link all the necessary libs and then execute the swift playground.

Reply:
also this should be beefed up to compile all of the playgrounds in the repo, but my shell-fu isn't good enough for that. may need some help

Reply:
Made it a bit more general. May need to beef up even more for source dependencies...

Reply:
nice thanks! 

yeah I don't think we need to worry about sources. I imagine anything we wanna put in there should prob just go in the library itself.

Reply:
We could also try something like this: https://flight.school/articles/running-xcode-playgrounds-on-travis-ci/

Reply:
yeah I looked into that. but also we could concat everything in the sources directory to the `Contents.swift` right? seems simpler